breacharbiter.go

```diff
@@ -30,6 +30,51 @@ import (
 // continue from the persisted state.
 var retributionBucket = []byte("retribution")
 
+// BreachConfig bundles the required subsystems used by the breach arbiter. An
+// instance of BreachConfig is passed to newBreachArbiter during instantiation.
+type BreachConfig struct {
+	// Signer is used by the breach arbiter to generate sweep transactions,
+	// which move coins from previously open channels back to the user's
+	// wallet.
+	Signer lnwallet.Signer
+
+	// DB provides access to the user's channels, allowing the breach
+	// arbiter to determine the current state of a user's channels, and how
+	// it should respond to channel closure.
+	DB *channeldb.DB
+
+	// Wallet facilitates sweep transaction publication and further
+	// facilitates the generation of receiver scripts when sweeping funds.
+	Wallet *lnwallet.LightningWallet
```

Comment:
The `Wallet` itself can similarly be omitted. From the PoV of the breach arb, the only functionality it needs from the wallet is `PublishTransaction.